org.lflang.tests/src/org/lflang/tests/runtime/CCppTest.java

```diff
@@ -33,18 +33,23 @@ public CCppTest() {
     @Test
     public void runAsCCpp() {
         Assumptions.assumeFalse(isWindows(), Message.NO_WINDOWS_SUPPORT);
-
-        EnumSet<TestCategory> categories = EnumSet.allOf(TestCategory.class);
-        categories.removeAll(EnumSet.of(
-            // Don't need to test examples.
-            // If any of them uses CCpp, it will
-            // be tested when compileExamples is
-            // run.
-            TestCategory.EXAMPLE));
-
-        runTestsForTargets(Message.DESC_AS_CCPP, categories::contains,
+        runTestsForTargets(Message.DESC_AS_CCPP, CCppTest::runAsCCppExclusion,
                            it -> ASTUtils.changeTargetName(it.fileConfig.resource,
                                                            Target.CCPP.getDisplayName()),
                            TestLevel.EXECUTION, true);
     }
+
+    /**
+     * Exclusion function for runAsCCpp test
+     */
+    static private boolean runAsCCppExclusion(TestCategory category) {
+        boolean excluded = false;
+        // Don't need to test examples.
+        // If any of them uses CCpp, it will
+        // be tested when compileExamples is
+        // run.
+        excluded |= (category == TestCategory.EXAMPLE);
+        excluded |= (!isLinux() && (category == TestCategory.DOCKER_FEDERATED || category == TestCategory.DOCKER_NONFEDERATED));
```

Comment:
The docker tests are excluded when the OS is non-linux. The reason for that is docker does not natively support Mac on Github action, and workaround solutions are unstable (see: https://github.community/t/why-is-docker-not-installed-on-macos/17017). Docker on Mac runs on a Linux virtual machine. I decided that it would be the simplest to just test on Linux.

Reply:
Ah I see. Yes I know about that limitation (see my comment below).
I think I misread this line of code.
Thanks for clarifying!